Report/sections/4_testing/unit_testing.tex

```diff
@@ -0,0 +1,106 @@
+\section{Unit Testing}
+In implementing unit testing, we have followed a few approaches.
+In general, we have attempted to follow the Act-Arrange-Assert (AAA)
+methodology, as well as the naming convention 
+\texttt{MethodName\_Scenario\_ExpectedResult}.
+
+We did not use Test-Driven Development, instead resorting to testing
+when we've finished the implementation of each part. While not ideal, it
+contributed to us catching bugs and fixing errors that we wouldn't
+otherwise have caught.
+
+To illustrate our approach to testing, we will draw examples from our
+AST Builder tests.
+
+Our approach is to traverse the AST through a visitor. First,
+we had the test fixture class inherit from an AST visitor interface.
+Then we implemented traversal between nodes, where each node knows how
+to visit the node(s) below.
+
+To test the visitor, we wrote some test code in \dazel{}, which can be seen in \snipref{lst:asttestcode}. We made sure that
```

Comment:
we wrote some test source code in \dazel{} which can be seen...

Reply:
Was supposed to say AST builder, not visitor. Ooops haha.